src/test/java/de/tum/in/www1/artemis/service/AuthorizationCheckServiceTest.java

```diff
@@ -0,0 +1,113 @@
+package de.tum.in.www1.artemis.service;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.time.ZonedDateTime;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.stereotype.Component;
+import org.springframework.test.context.TestPropertySource;
+
+import de.tum.in.www1.artemis.AbstractSpringIntegrationBambooBitbucketJiraTest;
+import de.tum.in.www1.artemis.domain.Course;
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.repository.CourseRepository;
+import de.tum.in.www1.artemis.web.rest.errors.AccessForbiddenException;
+
+class AuthorizationCheckServiceTest extends AbstractSpringIntegrationBambooBitbucketJiraTest {
+
+    private static final String TEST_PREFIX = "authorizationservice";
+
+    @BeforeEach
+    void initTestCase() {
+        database.addUsers(TEST_PREFIX, 2, 0, 0, 1);
+    }
+
+    @Nested
+    @Component
+    // The following annotation can only be applied to classes: https://github.com/spring-projects/spring-framework/issues/18951
+    // Only the login name of the student2 user is NOT allowed to self-register for courses.
+    @TestPropertySource(properties = "artemis.user-management.course-registration.allowed-username-pattern=^(?!" + TEST_PREFIX + "student2).*$")
+    class IsUserAllowedToSelfRegisterForCourseTest {
+
+        // We need our own courseService here that overshadows the one from the CourseServiceTest, so that the new property is applied to it.
+        @Autowired
+        private AuthorizationCheckService authCheckService;
+
+        @Autowired
+        private CourseRepository courseRepository;
+
+        private User student1;
+
+        private Course getCourseForSelfRegistrationAllowedTest() {
+            var course = database.createCourse();
+            course.setRegistrationEnabled(true);
+            course.setStartDate(ZonedDateTime.now().minusDays(2));
+            course.setEndDate(ZonedDateTime.now().plusDays(2));
+            course.setStudentGroupName("test-students");
+            return course;
+        }
+
+        @BeforeEach
+        void setUp() {
+            this.student1 = database.getUserByLogin(TEST_PREFIX + "student1");
+        }
+
+        @Test
+        @WithMockUser(username = TEST_PREFIX + "student1", roles = "USER")
+        void testIsUserAllowedToSelfRegisterForCourseForAllowed() {
+            Course course = getCourseForSelfRegistrationAllowedTest();
+            courseRepository.save(course);
+            assertDoesNotThrow(() -> authCheckService.checkUserAllowedToSelfRegisterForCourseElseThrow(this.student1, course));
```

Comment:
Hmm, I'm not entirely sure about this one. I think the code `assertThatCode(() -> authCheckService.checkUserAllowedToSelfRegisterForCourseElseThrow(this.student1, course)).doesNotThrowAnyException();` expresses the intended test better than just `authCheckService.checkUserAllowedToSelfRegisterForCourseElseThrow(this.student1, course)`. For the second version I would need an additional comment explaining where the test is, while the first version makes this clear immediately. Should I still change it, with a comment?

Reply:
I would leave out doesNotThrowAnyException() and comment the test case, but I think it would be a good idea to ask a more experienced developer. @julian-christl what do you think?

Reply:
I can get behind Paul's argument. In combination with the method named `...ElseThrow`, it's clear in my opinion.
Explicit is better than implicit.